src/StochasticStyles/StochasticStyles.jl

```diff
@@ -0,0 +1,56 @@
+"""
+This module defines [`StochasticStyle`](@ref)s, which tell Rimu which algorithm to use when
+doing stochastic matrix-vector multiplication.
+
+Exports:
+
+Interface functions for generalized vectors and Hamiltonians:
+
+* [`deposit!`](@ref)
+* [`diagonal_element`](@ref)
+* [`offdiagonals`](@ref)
+* [`random_offdiagonal`](@ref)
+* [`default_style`](@ref)
+
+Functions Rimu.jl uses to do FCIQMC:
+
+* [`fciqmc_col!`](@ref)
+* [`step_stats`](@ref)
+* [`update_dvec!`](@ref)
+
+Stochastic styles:
+
+* [`StochasticStyle`](@ref)
+* [`IsStochasticInteger`](@ref)
+* [`IsDeterministic`](@ref)
+* [`IsStochasticWithThreshold`](@ref)
+* [`IsDynamicSemistochastic`](@ref)
+* [`IsExplosive`](@ref)
+* [`StyleUnknown`](@ref)
+
+The offdiagonal spawning is defined in `spawning.jl` and is controlled by setting a
+[`SpawningStrategy`](@ref).
+
+The vector compression strategies are defined in `compression.jl` and are controlled by
+setting a [`CompressionStrategy`](@ref).
+
+"""
+module StochasticStyles
+
+using StaticArrays
+using StatsBase
+using ..ConsistentRNG
+using ..Rimu: MultiScalar, localpart
```

Comment:
How does this work? `StochasticStyles.jl` is included in `Rimu.jl`. This looks like a cyclic import.
Shouldn't it fail?

Reply:
Definitions and imports happen in order. Rimu defines `helpers.jl`, then all submodules get imported.
Another example of a dependency like this happens with `RMPI`.

Reply:
Thanks! I remember ...